pkg/topology/kademlia/kademlia_test.go

```diff
@@ -1646,6 +1646,93 @@ func TestAnnounceBgBroadcast(t *testing.T) {
 	}
 }
 
+func TestAnnounceNeighborhoodToNeighbor(t *testing.T) {
+	t.Parallel()
+
+	defer func(p int) {
+		*kademlia.SaturationPeers = p
+	}(*kademlia.SaturationPeers)
+	*kademlia.SaturationPeers = 4
+
+	defer func(p int) {
+		*kademlia.OverSaturationPeers = p
+	}(*kademlia.OverSaturationPeers)
+	*kademlia.OverSaturationPeers = 4
+
```

Comment:
Test which modify global variables shouldn't be executed in parallel because race condition may occur in which case one test function (executed in parallel) my affect expected variable of another test function.

PR https://github.com/ethersphere/bee/pull/3360 fixes hacky usage of global variables in kadeemlia (which makes possible for test to execute in parallel).

Reply:
@vladopajic Let's merge this, then can you also take a look at this test since your PR takes care of the global variables.